src/Resolution.test.ts

```diff
@@ -359,5 +362,30 @@ describe('Resolution', () => {
       expectSpyToBeCalled(eyes);
       expect(ethAddress).toBe('0x8aaD44321A86b170879d7A244c1e8d360c99DdA8');
     });
+
+    it('should work with web3HttpProvider', async () => {
+      const provider = new Web3HttpProvider(secretInfuraLink());
+      const resolution = Resolution.fromWeb3Provider(provider);
+      const eyes = mockAsyncMethods(resolution.cns, {
+        getResolver: '0xb66DcE2DA6afAAa98F2013446dBCB0f4B0ab2842',
+        getRecord: '0x8aaD44321A86b170879d7A244c1e8d360c99DdA8'
+      });
```

Comment:
These tests won't be able to test anything if we mock CNS functions directly - the provider flow isn't covered

Reply:
Wow that is so true. I start getting the issues with web3 providers after I remove these mocks.